### PR TITLE
Include latest changes in unit tests

### PR DESCRIPTION
**Pull request by JAlvarezJarreta**
Due to these bugs/additional features being blockers we merged them without requiring updates in the unit tests. This PR sorts that out to get back to a better code coverage.

I have taken the opportunity to review and update a few docstrings.

### CODE REVIEW

File: src/ensembl/utils/plugin.py [hunk 1]
```diff
@@ -194,6 +197,9 @@ def test_method(..., test_dbs: dict[str, UnitTestDB], ...):
         if src is not None:
             src = Path(src)
         name = argument.get("name", None)
-        key = name if name else src.name
+        try:
+            key = name if name else src.name
```
Comment (Dishalodha) [inline src/ensembl/utils/plugin.py:201]: curious if we can instead use or operator here 
```suggestion
            key = name or src.name
```
```suggestion
            key = name | src.name
```

File: tests/argparse/test_argparse.py [hunk 2]
```diff
@@ -307,3 +304,12 @@ def test_parse_args(self) -> None:
         # Check that the error is handled properly
         with raises(SystemExit):
             parser.parse_args(["--src_host", "host", "--src_port", "42", "--src_user", "username"])
+
+    @pytest.mark.dependency(depends=["add_server_arguments"])
+    def test_parse_args_not_server_host(self) -> None:
+        """Tests `ArgumentParser.parse_args()` method when a non-database server host is added as argument."""
```
Comment (Dishalodha) [inline tests/argparse/test_argparse.py:310]: Just to understand more clearly that this test matches based on prefix, do we want to add another parameter to test what happens then the arg is like 'file_hostdb' ?
```suggestion
        """Tests `ArgumentParser.parse_args()` method when a non-database server host is added as argument.
        This test checks that arguments with a `host` suffix in their names (e.g., `file_host`) are parsed correctly. 

        """
```

**Review by Dishalodha — COMMENTED** [review]
Looks good, thank you!

File: src/ensembl/utils/plugin.py (hunk 1 above)
Comment (JAlvarezJarreta) [reply src/ensembl/utils/plugin.py:201]: This is a new trick I didn't know: it seems `|` does not work if either is `None`, but `or` does!

File: tests/argparse/test_argparse.py [hunk 3]
```diff
@@ -307,7 +308,11 @@ def test_parse_args_url_arg_present(self) -> None:
 
     @pytest.mark.dependency(depends=["add_server_arguments"])
     def test_parse_args_not_server_host(self) -> None:
-        """Tests `ArgumentParser.parse_args()` method when a non-database server host is added as argument."""
+        """Tests `ArgumentParser.parse_args()` method when a non-database server host is added as argument.
+
+        This test checks that arguments with a `host` suffix in their names (e.g., `file_host`) are parsed correctly.
```
Comment (JAlvarezJarreta) [inline tests/argparse/test_argparse.py:313]:
```suggestion
        This test checks that arguments with a `host` suffix in their names (e.g. `file_host`) are
        parsed correctly.
```

**Review by Dishalodha — APPROVED** [review]

**Pull request merged**